### ISSUE TITLE
opts.path interferes with resolved query caching

### ISSUE THREAD

**Issue by thoughtspile** [browserslist/browserslist#860]
When debugging our build, I've noticed that even after forcing `BROWSERSLIST_CONFIG` as a workaround for #859 , browserslist still takes ~1s to run. About half that time is spent on [repeatedly resolving](https://github.com/browserslist/browserslist/blob/0639f5484b46b6c8608eab13a22310e9922feb96/index.js#L424) our single browserslist config into browsers:

<img width="1053" alt="Снимок экрана 2024-12-14 в 17 28 08" src="https://github.com/user-attachments/assets/53a56bf1-9c30-4c66-9bed-696f9eca979e" />

From the looks of it, the resolution cache key includes `opts.path`, which, in our case (vite -> autoprefixer -> browserslist) equals CSS file path:

https://github.com/browserslist/browserslist/blob/0639f5484b46b6c8608eab13a22310e9922feb96/index.js#L404C1-L421C52

However, nothing in `resolve` seems to reference `path` (or any `context` field other than `mobileToDesktop`, for that matter. 

Could we maybe exclude `path` from `cacheKey`? Together with the #859 workaround, this brings browserslist time to a reasonable 400ms:

<img width="1094" alt="Снимок экрана 2024-12-14 в 17 35 09" src="https://github.com/user-attachments/assets/53cccfd4-07fc-4a75-9291-dd903a74ddb8" />

**Comment by ai** [comment]
> However, nothing in resolve seems to reference path (or any context field other than mobileToDesktop, for that matter.

No. Browserslist is looking for Browserslist config closest to the file.

https://github.com/browserslist/browserslist/blob/main/node.js#L258-L259

The original idea is to allow having different browsers depending on file path:

```
src/
  legacy/
    .browserslistrc
    index.css // Legacy browsers
  index.css // Modern browsers
.browserslistrc
```

Maybe we can change it in next major, but not before it.

Maybe we can add env var to disable realtive path search?

**Comment by ai** [comment]
Cache is using here https://github.com/browserslist/browserslist/blob/main/node.js#L392-L394

**Comment by thoughtspile** [comment]
> Cache is using here https://github.com/browserslist/browserslist/blob/main/node.js#L392-L394

browserslist has multile cache layers, in this issue let's focus on the one in `index.js`

https://github.com/browserslist/browserslist/blob/0639f5484b46b6c8608eab13a22310e9922feb96/index.js#L421-L422

it caches query resolution, e.g. 

```js
queries = ["last 2 years AND >.05%","ios >= 13","not IE 11","not op_mini all","not dead"]
context = {"path":"/Users/vklepov/project/src/cryptopro/+page.svelte.vite-preprocess.postcss"}
```

into

```js
['and_chr 129', 'and_ff 130', 'and_qq 14.9', 'and_uc 15.5', 'android 129', 'chrome 128', ...etc]
```

On this layer, including `path` in `cacheKey` makes the cache largely non-functional, because each CSS file generates a unique cache key.

**Comment by thoughtspile** [comment]
Here's a PR to better illustrate the idea
